packages/react/src/components/form/Textfield/Textfield.tsx

```diff
@@ -1,217 +1,99 @@
-import { PadlockLockedFillIcon } from '@navikt/aksel-icons';
-import cl from 'clsx/lite';
-import type { InputHTMLAttributes, ReactNode } from 'react';
-import { forwardRef, useId, useState } from 'react';
+import {
+  type ForwardedRef,
+  Fragment,
+  type InputHTMLAttributes,
+  type ReactNode,
+  type TextareaHTMLAttributes,
+  forwardRef,
+} from 'react';
 
 import type { DefaultProps } from '../../../types';
-import { omit } from '../../../utilities';
 import { Label } from '../../Label';
-import { Paragraph } from '../../Paragraph';
 import { ValidationMessage } from '../../ValidationMessage';
-import type { CharacterLimitProps } from '../CharacterCounter';
-import { CharacterCounter } from '../CharacterCounter';
-import type { FormFieldProps } from '../useFormField';
+import {
+  Field,
+  FieldAffix,
+  FieldAffixWrapper,
+  FieldDescription,
+} from '../Field';
+import { Input } from '../Input';
+import { Textarea } from '../Textarea';
 
-import { useTextfield } from './useTextfield';
-
-export type TextfieldProps = {
+type SharedTextfieldProps = {
   /** Label */
   label?: ReactNode;
-  /** Visually hides `label` and `description` (still available for screen readers)  */
-  hideLabel?: boolean;
-  /** Prefix for field. */
+  /** Description */
+  description?: ReactNode;
+  /** Prefix */
   prefix?: string;
-  /** Suffix for field. */
+  /** Suffix */
   suffix?: string;
-  /** Supported `input` types */
-  type?:
-    | 'date'
-    | 'datetime-local'
-    | 'email'
-    | 'file'
-    | 'month'
-    | 'number'
-    | 'password'
-    | 'search'
-    | 'tel'
-    | 'text'
-    | 'time'
-    | 'url'
-    | 'week';
-  /**
-   *  The characterLimit function calculates remaining characters based on `maxCount`
-   *
-   *  Provide a `label` function that takes count as parameter and returns a message.
-   *
-   *  Use `srLabel` to describe `maxCount` for screen readers.
-   *
-   *  Defaults to Norwegian if no labels are provided.
-   */
-  characterLimit?: CharacterLimitProps;
-  /** Exposes the HTML `size` attribute.
-   * @default 20
-   */
-  htmlSize?: number;
-} & FormFieldProps &
-  InputHTMLAttributes<HTMLInputElement> &
-  DefaultProps;
+  /** Validation message. This will flag an error & assign `aria-invalid` */
+  validation?: ReactNode;
+} & DefaultProps;
+
+type TextareaTypes = {
+  /** Use to render a `textarea` instead of `input` for multiline support  */
+  multiline: true;
+} & TextareaHTMLAttributes<HTMLTextAreaElement>;
+
+type InputTypes = {
+  /** Use to render a `textarea` instead of `input` for multiline support  */
+  multiline?: never | false;
+} & InputHTMLAttributes<HTMLInputElement>;
```

Comment:
```suggestion
} & Omit<InputHTMLAttributes<HTMLInputElement>, 'prefix'>;
```

Reply:
Will update to InputProps so it matches!